### ISSUE TITLE
Cannot download large amounts of state on custom RPC node

### ISSUE THREAD

**Issue by ChaoticTempest** [near/nearcore#7883]
**Describe the bug**
When trying to `view_state` into a large contract like `ref.finance` with my own custom RPC node (to get beyond the 50kb limit), it throws me a `Error: Node Does Not Track The Shard ID 0` even though my `config.json` has `tracked_shards: [0]`.

Additionally, this could be a different bug, but when SIGINTing the RPC node with `ctrl+c`, it dies properly with all the cleanup messages, but after starting the node up again, it throws `DBNotFoundError`. Here's the full stack trace for it:
```
❯ RUST_BACKTRACE=full ./neard --home ~/.near run
2022-10-18T18:44:56.053238Z  INFO neard: version="1.29.0" build="1.29.0" latest_protocol=56
2022-10-18T18:44:56.144850Z  INFO db: Created a new RocksDB instance. num_instances=1
2022-10-18T18:44:56.147169Z  INFO db: Dropped a RocksDB instance. num_instances=0
2022-10-18T18:44:56.147181Z  INFO near: Opening RocksDB database path=/Users/tensor/.near-mainnet/data
2022-10-18T18:44:56.225081Z  INFO db: Created a new RocksDB instance. num_instances=1
2022-10-18T18:44:56.282306Z  INFO near_network::peer_manager::peer_manager_actor: Bandwidth stats total_bandwidth_used_by_all_peers=0 total_msg_received_count=0 max_max_record_num_messages_in_progress=0
2022-10-18T18:44:56.288218Z  INFO stats: #76165950 Waiting for peers 0 peers ⬇ 0 B/s ⬆ 0 B/s 0.00 bps 0 gas/s CPU: 0%, Mem: 77.1 MB
2022-10-18T18:44:56.288482Z DEBUG stats: EpochId(`7STQ9NZ61UCw6ec4DYoWRAuhUKFjLsq6nAaZT4f3bWpy`) Orphans: 0 With missing chunks: 0 In processing 0
thread 'actix-rt|system:0|arbiter:0' panicked at 'called `Result::unwrap()` on an `Err` value: Chain(DBNotFoundErr("C8nBVdj6u8hJV1NtTCYZZ5gPP8nKu9t5gKe4HtGSXcNm"))', chain/client/src/client_actor.rs:254:47
stack backtrace:
   0:        0x10360125c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::ha9a1a8bf7485458d
   1:        0x102c1ceb8 - core::fmt::write::h9ee5b099821ae5e1
   2:        0x1035db6ec - std::io::Write::write_fmt::h240449cbf2a536ee
   3:        0x1036051a8 - std::panicking::default_hook::{{closure}}::h321649ccf26de565
   4:        0x103605e4c - std::panicking::rust_panic_with_hook::hd083a3aa5c934ce6
   5:        0x103605a40 - std::panicking::begin_panic_handler::{{closure}}::hb5be8aaa10a229ca
   6:        0x1036059b8 - std::sys_common::backtrace::__rust_end_short_backtrace::h68e10e4f00198298v
   7:        0x103605984 - _rust_begin_unwind
   8:        0x103c74b04 - core::panicking::panic_fmt::hffc63a015c61fdde
   9:        0x103c74cbc - core::result::unwrap_failed::hd24fd75dfe8b9563
  10:        0x102e96b10 - <actix::contextimpl::ContextFut<A,C> as core::future::future::Future>::poll::hb4e0bb84fa7a24c1
  11:        0x102e94034 - tokio::runtime::task::raw::poll::hb94478d1e0b97bd7
  12:        0x1036281d4 - tokio::task::local::LocalSet::tick::hb2e2950095e24289
  13:        0x102b9eac8 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h9823e34511e40eee
  14:        0x102b9cd24 - std::sys_common::backtrace::__rust_begin_short_backtrace::hc5a94bc37ad08a88
  15:        0x102b9bc10 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h54248d377328669a
  16:        0x1036071a0 - std::sys::unix::thread::Thread::new::thread_start::hacb0bdc43acdb831
  17:        0x19c12426c - __pthread_deallocate
zsh: abort      RUST_BACKTRACE=full ./neard-1.29.0 --home ~/.near-mainnet run
```
This DB corruption seems to only happen after trying to download large amounts of state with `view_state`. This is annoying to run into, because I'm not sure how to get it back into a good state besides redownloading the whole chain from AWS (which is like 500gb at this point).

**To Reproduce**
- Setup neard as usual with 1.29.0 tag branch and `make release`. (also tested with master branch but same result)
- Set `"trie_viewer_state_size_limit": 10000000000,` in the `config.json` in the network's home folder
- Run the node as usual: `neard --home {home_folder} run`
- `near view-state v2.ref-finance.near --finality final --node_url http://localhost:3030`

**Expected behavior**
Should be able to download state (albeit having to wait a long time since it's gigabytes of data).

**Version (please complete the following information):**
- nearcore 1.29.0 or master
- rust version 1.64
- mainnet

**Additional context**
Other people have ran into similar issues with trying to download large amounts of state. @itegulov if you want to dump your findings here as well. There was a discussion on [zulip](https://near.zulipchat.com/#narrow/stream/295302-general/topic/near.20view-state.20error/near/287929319) but that didn't seem to lead anywhere.

**Comment by ChaoticTempest** [comment]
Seems this is occurring only for macos M1 as @itegulov got it working for a contract the size of 60mb but on linux

**Comment by akhi3030** [comment]
@mina86, @Longarithm: AFAIK, you two have a lot of familiarity with state viewer.  Can you PTAL?  Please apply the appropriate labels (for which teams should be involved, etc.)

**Comment by Longarithm** [comment]
After a brief look, "Error: Node Does Not Track The Shard ID 0" seem to be the most important here. @ChaoticTempest could you provide a stacktrace for this error?

I remember some issues in the past related to `tracked_shards` config option and perhaps it should be actually set to `tracked_shards: [0, 1, 2, 3]`. @mzhangmzz do you know if it is the case and if it can cause DB corruption?

It is also weird that the error is reproducible on macos and not reproducible on linux, so I am not sure if core or node teams should be involved. In fact, `view_state` shouldn't affect DB at all.

**Comment by akhi3030** [comment]
If we need access to a Pagoda engineer with Macs, @Ekleog-NEAR has a M1 Macbook.  We could ask him to run experiments if needed.

**Comment by ChaoticTempest** [comment]
@Longarithm this was an RPC returned error, so there was no stacktrace associated with it. But looking through nearcore code, that error message is associated with `QueryError::UnavailableShard` which gets created only in the following snippet from what I've found: https://github.com/near/nearcore/blob/d7626c0be46912c3ac9000a9e35c2eb0982749c4/chain/client/src/view_client.rs#L280-L292

**Comment by mzhangmzz** [comment]
@ChaoticTempest yes your setup is right, so the machine shouldn't return the "UnavailableShard" error. When you submit the rpc, did it return immediately with this error? The code path to return this error happens before the state was being queried, so I think your database might have already been corrupted even before this query.

**Comment by ChaoticTempest** [comment]
It took roughly like 15 minutes before returning me the result for the first query. Meanwhile, subsequent queries were immediate which could mean the DB would have been corrupted by the first query.

**Comment by mzhangmzz** [comment]
@ChaoticTempest what is the result of the first query?

**Comment by ChaoticTempest** [comment]
it's the same error message with `The node does not track the shard ID 0`. All the queries are the same, but sometimes it can be different shard ids like `The node does not track the shard ID 3`
